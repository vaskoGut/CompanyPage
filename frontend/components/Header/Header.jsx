import { useQuery } from 'graphql-hooks';
import Link from '@/components/Link';
import { x } from '@xstyled/styled-components';
import { GetMainNav } from '@/lib/queries';
import SHeader from './styles';

const Header = () => {
  const { loading, error, data } = useQuery(GetMainNav, {});
  const { menuItems: { nodes: menuItems } = {} } = data || {};
  if (error) return null;

  return (
    <SHeader>
      {data?.pageBy?.title}
      {menuItems && !!menuItems.length > 0 && (
        <nav>
          <x.div display="flex" p={0} as="ul" justifyContent="center" mb={20}>
            {menuItems.map((menuItem) => (
              <x.li mx={10} key={menuItem.id}>
                <Link
                  uri={menuItem.uri ? menuItem.uri : undefined}
                  url={menuItem.url}
                  title={menuItem?.title ? menuItem.title : undefined}
                  rel={menuItem?.linkRelationship ? menuItem.linkRelationship : undefined}
                  target={menuItem.target ? menuItem.target : undefined}
                >
                  {menuItem.label}
                </Link>
              </x.li>
            ))}
            <x.li mx={10}>
              <Link uri="/search" url="search">
                Search
              </Link>
            </x.li>
          </x.div>
        </nav>
      )}
    </SHeader>
  );
};

export default Header;
