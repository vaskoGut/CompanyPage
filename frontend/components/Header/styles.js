import styled from '@xstyled/styled-components';

const SHeader = styled.header`
  ul {
    list-style: none;
  }
`;

export default SHeader;
