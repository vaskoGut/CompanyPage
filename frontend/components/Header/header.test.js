import React from 'react';
import { shallow } from 'enzyme';
jest.mock('graphql-hooks');
import { useQuery } from 'graphql-hooks';
import Header from './Header';

useQuery.mockReturnValue({
  data: {
    loading: false,
    error: null,
    menuItems: {
      nodes: [
        {
          id: '1',
          name: 'Overig (algemene RI&E)',
        },
        {
          id: '2',
          name: 'Bakkers',
        },
        {
          id: '3',
          name: 'Kappers',
        },
        {
          id: '4',
          name: 'Schoenherstellers',
        },
        {
          id: '5',
          name: 'Slagers',
        },
      ],
    },
  },
});

describe('Header', () => {
  it('should render correctly with children', async () => {
    const element = shallow(<Header />);
    expect(element).toMatchSnapshot();
  });
});
