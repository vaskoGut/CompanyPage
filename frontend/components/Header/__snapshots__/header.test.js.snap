// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly with children 1`] = `
<styled.header>
  <nav>
    <styled.div
      as="ul"
      display="flex"
      justifyContent="center"
      mb={20}
      p={0}
    >
      <styled.li
        key="1"
        mx={10}
      >
        <Link />
      </styled.li>
      <styled.li
        key="2"
        mx={10}
      >
        <Link />
      </styled.li>
      <styled.li
        key="3"
        mx={10}
      >
        <Link />
      </styled.li>
      <styled.li
        key="4"
        mx={10}
      >
        <Link />
      </styled.li>
      <styled.li
        key="5"
        mx={10}
      >
        <Link />
      </styled.li>
      <styled.li
        mx={10}
      >
        <Link
          uri="/search"
          url="search"
        >
          Search
        </Link>
      </styled.li>
    </styled.div>
  </nav>
</styled.header>
`;
